### PR TITLE
[enhence](memory) gc inverted index cache when there is not enough memory

### PR DESCRIPTION
**Pull request by ZhangYu0123**
# Proposed changes

Support to gc inverted index cache when there is not enough memory. 
(1) problem： The inverted index cache (InvertedIndexSearcherCache and InvertedIndexQueryCache) may use 20% memory which can't be released. 

(2) effect after gc：
image

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ZhangYu0123** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index_cache.cpp [hunk 1]
```diff
@@ -179,6 +179,19 @@ Status InvertedIndexSearcherCache::erase(const std::string& index_file_path) {
     return Status::OK();
 }
 
+void InvertedIndexSearcherCache::prune() {
+    if (_cache) {
+        _cache->prune();
```
Comment (xinyiZzz) [inline be/src/olap/rowset/segment_v2/inverted_index_cache.cpp:184]: `prune()` will free the entire cache, expecting it to be like `segment cache`,
During minor gc, prune_if evict the cache value with last_visit_time > N min
During full gc, prune evict entire cache

**Comment by ZhangYu0123** [comment]
run buildall

File: be/src/util/mem_info.cpp [hunk 2]
```diff
@@ -113,6 +114,16 @@ void MemInfo::process_cache_gc(int64_t& freed_mem) {
                 StoragePageCache::instance()->get_page_cache_mem_consumption(segment_v2::DATA_PAGE);
         StoragePageCache::instance()->prune(segment_v2::DATA_PAGE);
     }
+
+    if (segment_v2::InvertedIndexSearcherCache::instance()->mem_consumption() > min_free_size) {
+        freed_mem += segment_v2::InvertedIndexSearcherCache::instance()->mem_consumption();
```
Comment (xinyiZzz) [inline be/src/util/mem_info.cpp:119]: `mem_consumption` is all cache memory, but here only the expired value is released.

can let InvertedIndexSearcherCache -> prune return evict size, or do not add freed_mem here

**Comment by ZhangYu0123** [comment]
run p1

**Comment by ZhangYu0123** [comment]
run p1

**Review by xinyiZzz — DISMISSED** [review]
LGTM

**Comment by ZhangYu0123** [comment]
run buildall

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**